addOns/graphql/src/main/java/org/zaproxy/addon/graphql/GraphQlParser.java

```diff
@@ -162,6 +177,35 @@ private void generate(String schema) {
         }
     }
 
+    private void raiseIntrospectionAlert(HttpMessage msg) {
+        var extAlert =
+                Control.getSingleton().getExtensionLoader().getExtension(ExtensionAlert.class);
+        if (extAlert == null) {
+            return;
+        }
+        Alert alert =
+                Alert.builder()
+                        .setPluginId(ExtensionGraphQl.TOOL_ALERT_ID)
+                        .setAlertRef(INTROSPECTION_ALERT_REF)
```

Comment:
I believe on other rules when there’s only a single alert possible we’ve skipped setting the ref.

Reply:
Since the source of this alert is a TOOL, as per discussions in IRC, the idea is that each non-scan-rule alert raised by the tool could have its own ref.

Reply:
Okay, sounds good.